builtin/providers/aws/resource_aws_config_configuration_recorder.go

```diff
@@ -0,0 +1,145 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/configservice"
+)
+
+func resourceAwsConfigConfigurationRecorder() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsConfigConfigurationRecorderPut,
+		Read:   resourceAwsConfigConfigurationRecorderRead,
+		Update: resourceAwsConfigConfigurationRecorderPut,
+		Delete: resourceAwsConfigConfigurationRecorderDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "default",
+				ValidateFunc: validateMaxLength(256),
+			},
+			"role_arn": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
Because API docs are (unfortunately) wrong. If you try creating the recorder without Role ARN, you'll get the following error:

```
Creating Configuration Recorder failed: InvalidRoleException: The role arn passed 'null' is not valid.
	status code: 400, request id: dd0f8d90-ee06-11e6-8688-c1e7da7e9aca
```

and in case you try to pass an empty string you'll get this one:
```
Creating Configuration Recorder failed: InvalidRoleException: The role arn passed '' is not valid.
	status code: 400, request id: a4eabbda-ee06-11e6-922d-b1882f744aba
```
so there doesn't seem to be a way to leave it out.

We should perhaps report that to Amazon.